components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java

```diff
@@ -96,11 +106,18 @@ public KubernetesDeploymentProvider() {
     /**
      * Get list of deployments.
      *
+     * @param namespace Namespace of the tenant
      * @return Array of deployments
      */
     @Override
-    public List<Deployment> listDeployments() {
-        return kubernetesClient.extensions().deployments().list().getItems().stream()
+    public List<Deployment> listDeployments(String namespace) {
+        DeploymentList list = kubernetesClient.extensions().deployments().inNamespace(namespace)
+                .withLabel(WSO2_DEPLOYMENT_LABEL).list();
+        if (list == null || list.getItems() == null || list.getItems().isEmpty()) {
+            return new ArrayList<>();
```

Comment:
Does the caller of this method expects to add new elements to the returned List. If not we can use Collections.emptyList();